src/auth/src/credentials.rs

```diff
@@ -165,3 +165,107 @@ pub mod traits {
         }
     }
 }
+
+/// The path to Application Default Credentials (ADC), as specified in [AIP-4110].
+///
+/// [AIP-4110]: https://google.aip.dev/auth/4110
+#[allow(dead_code)] // TODO(#442) - implementation in progress
+fn adc_path() -> Option<String> {
+    if let Ok(e) = std::env::var("GOOGLE_APPLICATION_CREDENTIALS") {
+        return Some(e);
+    }
+    adc_well_known_path()
+}
+
+/// The well-known path to ADC on Windows, as specified in [AIP-4113].
+///
+/// [AIP-4113]: https://google.aip.dev/auth/4113
+#[cfg(target_os = "windows")]
+fn adc_well_known_path() -> Option<String> {
+    if let Ok(root) = std::env::var("APPDATA") {
+        return Some(root + "/gcloud/application_default_credentials.json");
+    }
+    None
+}
+
+/// The well-known path to ADC on Linux and Mac, as specified in [AIP-4113].
+///
+/// [AIP-4113]: https://google.aip.dev/auth/4113
+#[cfg(not(target_os = "windows"))]
+fn adc_well_known_path() -> Option<String> {
+    if let Ok(root) = std::env::var("HOME") {
+        return Some(root + "/.config/gcloud/application_default_credentials.json");
+    }
+    None
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use scoped_env::ScopedEnv;
+
+    #[cfg(target_os = "windows")]
+    #[test]
+    #[serial_test::serial]
+    fn adc_well_known_path_windows() {
+        let _creds = ScopedEnv::remove("GOOGLE_APPLICATION_CREDENTIALS");
+        let _appdata = ScopedEnv::set("APPDATA", "C:/Users/foo");
+        assert_eq!(
+            adc_well_known_path().unwrap(),
+            "C:/Users/foo/gcloud/application_default_credentials.json"
+        );
+        assert_eq!(
+            adc_path().unwrap(),
+            "C:/Users/foo/gcloud/application_default_credentials.json"
+        );
+    }
+
+    #[cfg(target_os = "windows")]
+    #[test]
+    #[serial_test::serial]
+    fn adc_well_known_path_windows_no_appdata() {
+        let _creds = ScopedEnv::remove("GOOGLE_APPLICATION_CREDENTIALS");
+        let _appdata = ScopedEnv::remove("APPDATA");
+        assert!(adc_well_known_path().is_none());
```

Comment:
Here and below, consider:

```suggestion
        assert_eq!(adc_well_known_path(), None);
```

which should give you better error messages on failure.

Reply:
Fixed